mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -3127,93 +3261,169 @@ fn lp_connected_alice(ctx: MmArc, taker_order: TakerOrder, taker_match: TakerMat
             uuid
         );
 
-        let now = now_sec();
-        if ctx.use_trading_proto_v2() {
-            let taker_secret = match generate_secret() {
-                Ok(s) => s.into(),
-                Err(e) => {
-                    error!("Error {} on secret generation", e);
-                    return;
-                },
+        let bob_swap_v = taker_match.reserved.swap_version;
+        // Start legacy swap if maker uses legacy protocol (version 1) or if conditions for trading_proto_v2 aren't met.
+        if bob_swap_v == 1u32 || !(ctx.use_trading_proto_v2() && bob_swap_v == 2u32) {
```

Comment:
the part `bob_swap_v == 2` is redundant.

Reply:
could you clarify why?
I decided to have such explicit check/restriction as we are going support only legacy and v2 versions for now.

Reply:
if `bob_swap_v` is not `1` then it must be `2`!

Reply:
> if `bob_swap_v` is not `1` then it must be `2`!

what if other side provided changed configurations? I mean if they changed code with swap version. Thats almost impossible situation, but as right now version must be either 1 or 2 I added such explicit restriction. When we have more versions, we can change it to min version check.

Reply:
> what if other side provided changed configurations? I mean if they changed code with swap version.

that doesn't do the approach above no good. if `bob_swap_v` was `3`, you end up with `!(use_proto_v2() && bob_swap_v == 2) -> !(true && false) -> true`, which will initiate a legacy swap and i don't think this is what you want.

That said, if we get a higher version this means the other node is more updated which usually has the responsibility to handle the backward compatibility cases. In here you resort to legacy when the other node is updated as a means of backward compatibility (is it forward in this scenario?!!), but in fact the updated node is the one that should choose to be backward compatible with us and not the opposite.

The check can simply be: `if i_use_v1 || they_use_v1 { start_legacy() }` -> `bob_swap_v == 1 || !ctx.use_trading_proto_v2()`

Reply:
> which will initiate a legacy swap and i don't think this is what you want

That what I suggest to have while tpu is on alpha stage

Reply:
you are just forcing our future selves to transact over legacy with nodes that already support tpv2.
we shouldn't have any control over higher versions. higher versions should get the say of whether to reject us or not.
for e.g., would you ever do a check `if other_node_version > 5 { do_breaking_change() // breaking change is like tpu vs legacy }` where we haven't even reached version `5`!! of course not, you are restraining your later self for absolute no reason.

> That what I suggest to have while tpu is on alpha stage

it's not a **while**. once this is deployed it is deployed and whoever is running it will play with that logic. changing this logic in itself is a breaking change and will need a version bump.

Reply:
fixed https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/d3e6ae140c6866ea7bac661806e8cf3b2df28faa